src/error.rs

```diff
@@ -0,0 +1,25 @@
+//! Contains error types and other error handling tools.
+
+pub use crate::parse::ParseIntError;
+
+/// Impls std::error::Error for the specified type with appropriate attributes, possibly returning
+/// source.
+#[macro_export]
+macro_rules! impl_std_error {
```

Comment:
There are a bunch of other struct errors that we can use this for too
- ConversionError
- SighashTypeParseError
- CommandStringError
- ParseLengthError

I did all of these, I'll pushed a branch `review-parse-int-error` on my fork in case you want to look at it like that.

Reply:
Yeah, I was planning to do these in a followup PR.